walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,49 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<String> attendees = request.getAttendees();
+    
+    // build the list of all events to consider while picking a meeting range 
+    ArrayList<TimeRange> allEventRanges = new ArrayList<TimeRange>();
+    for (Event event : events) {
+      // consider the event as long as there is at least one attendee atending it 
+      if (!Collections.disjoint(event.getAttendees(), attendees)) {
+        allEventRanges.add(event.getWhen());
+      }
+    }
+    Collections.sort(allEventRanges, TimeRange.ORDER_BY_START); // sort chronologically 
+
+    // compile list of event times with no overlap
+    Collection<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    int eventEndTime = TimeRange.START_OF_DAY;   
+    for (TimeRange time : allEventRanges) {
+      int eventDuration = time.start() - eventEndTime; 
```

Comment:
What happens if I have two events that are

Event 1:  5 PM - 8 PM
Event 2:  6 PM - 7 PM

You look at event 1 and set eventEndTime to 8PM
Then you look at event 2 and calculate (6 - 8) = -2
Seems to me you should just skip any event that occurs earlier than your eventEndTime to make this possibility explicit.